fava/api/__init__.py

```diff
@@ -295,7 +302,33 @@ def interval_balances(self, interval, account_name, accumulate=False):
                 end_date, min_accounts=account_names)
             for begin_date, end_date in interval_tuples]
 
-        return zip_real_accounts(interval_balances), interval_tuples
+        return self.add_budgets(zip_real_accounts(interval_balances),
+                                interval_tuples, accumulate), interval_tuples
+
+    def add_budgets(self, zipped_interval_balances, interval_tuples, accumulate):  # noqa
+        """Add budgets data to zipped (recursive) interval balances."""
+        if not zipped_interval_balances:
+            return
+
+        interval_budgets = [self.budgets.budget(
+                zipped_interval_balances['account'],
+                interval_tuples[0][0] if accumulate else begin_date,
+                end_date - rdelta(days=1)
```

Comment:
Does this mean that the dates that the budget function accepts are inclusive? If so it should be changed to exclusive as that's the way the current code and beancount does it.

Also: a simple `timedelta(1)` can be used here.